models/ed/R/write.configs.ed.R

```diff
@@ -222,6 +222,24 @@ write.config.ED2 <- function(trait.values, settings, run.id, defaults=settings$c
   ed2in.text <- gsub('@END_DAY@', format(enddate, "%d"), ed2in.text)
   ed2in.text <- gsub('@END_YEAR@', format(enddate, "%Y"), ed2in.text)
   
+  ##-----------------------------------------------------------------------
+  #Set The flag for IMETAVG telling ED what to do given how input radiation was originally averaged
+  # -1 = I don't know, use linear interpolation
+  # 0 = No average, the values are instantaneous 
+  # 1 = Averages ending at the reference time
+  # 2 = Averages beginning at the reference time
+  # 3 = Averages centered at the reference time
+  
+  if (settings$run$inputs$met$source == "NARR") {
+    ed2in.text <- gsub('@MET_SOURCE@', 2,ed2in.text)
+  }else if (settings$run$inputs$met$source == "Ameriflux"){
+    ed2in.text <- gsub('@MET_SOURCE@', 2,ed2in.text)
+  }else if (settings$run$inputs$met$source == "CRUNCEP"){
+    ed2in.text <- gsub('@MET_SOURCE@', 2,ed2in.text)
+  }else{
+    ed2in.text <- gsub('@MET_SOURCE@', -1,ed2in.text)    
+  }
```

Comment:
More generally, the proposed solution is not acceptable. You are setting the interpolation scheme based on which data source was provided, which means that this code will need to be updated every single time a new met product is produced (and it notably only covers a subset of the  met we already support).  The interpolation needs to be set based on the TIMESTEP of the data, not the source. That said, this code seems to set the interpolation to 2 for all data sources regardless of their timestep -- and if that's the case then why bother doing this? Wouldn't it just be simpler to change the default in the template?